internal/upgrade/v0_6_0.go

```diff
@@ -0,0 +1,17 @@
+package upgrade
+
+import (
+	"context"
+
+	"github.com/grafana/tempo-operator/apis/tempo/v1alpha1"
+)
+
+// This upgrade unsets the image fields in the TempoStack CR.
+// From 0.6.0 onwards, the image location is not stored in the CR unless it got changed manually.
+func upgrade0_6_0(ctx context.Context, u Upgrade, tempo *v1alpha1.TempoStack) (*v1alpha1.TempoStack, error) {
+	tempo.Spec.Images.Tempo = ""
```

Comment:
If I understand this correctly, this is going to unset all the images on the CRs, but if an user specified an image and then do the upgrade, he will lose that setting?

Reply:
Thinking more about it. If the user configures a custom image, let's say a rebuild with modifications of tempo 2.2.3, and then upgrades the tempo operator and the new operator version requires tempo version 2.4.
This will likely break the installation anyway?

Reply:
In other words, setting a custom image only makes sense if the `TempoStack` is set to unmanaged imho (unmanaged also means it won't get upgraded).

Reply:
I don't think it is a way, I think the approach here is the right one. May be we need to put in the changelog or in some place that if you upgrade from 0.5.0 to 0.6.0 and you have custom images those changes will be overridden.

Reply:
Yes, that make sense to me.

Reply:
+1, I updated the text in the changelog entry to make it more clear.